TelemetryManager/Inc/MavlinkDecoder.hpp

```diff
@@ -0,0 +1,78 @@
+//
+// Created by Yarema Dzulynsky on 2023-07-10.
+//
+
+#ifndef MAVLINKDECODER_H
+#define MAVLINKDECODER_H
+
+#include <functional>
+// #include <iostream>
+#include <unordered_map>
+
+#include "Official_Mavlink_2_Library/common/mavlink.h"
+
+
+
+/**
+ * Macro for registering a decoder function for a specific message type.
+ * @param msgId (int) - the message id of the message type to be decoded.
+ * @param baseName (string) - the base name of the message type to be decoded. Ex. for
+ * mavlink_attitude_t, the base name is attitude. Essentially, whatever is in between mavlink_ and
+ * _t form the message you would like to register. The base name is used to generate the decoder
+ * function name. Ex. for mavlink_attitude_t, the decoder function name is
+ * mavlink_msg_attitude_decode.
+ * @param postProcessor ([](mavlink_**baseName**_t &message)) - the post processor function to be
+ * called after the message. This function must take in a reference to the message type that is
+ * being decoded. You can then use this reference to access the message fields. Ex. message.roll,
+ * message.pitch, etc. Now do what you want with them! :)
+ * @example REGISTER_DECODER(MAVLINK_MSG_ID_ATTITUDE, attitude, [](mavlink_attitude_t &message) {
+ *     //print out the message
+ *     std::cout << "ATTITUDE" << std::endl;
+ *     std::cout << "roll: " << message.roll << std::endl;
+ *     std::cout << "pitch: " << message.pitch << std::endl;
+ *     std::cout << "yaw: " << message.yaw << std::endl;
+ *     std::cout << "rollspeed: " << message.rollspeed << std::endl;
+ *     std::cout << "pitchspeed: " << message.pitchspeed << std::endl;
+ *     };
+ */
+#define REGISTER_DECODER(msgId, baseName, postProcessor)    \
```

Comment:
Great suggestion! Originally, I thought we would need to decode every type of message as it came in, so tried to make the code as dynamic as possible. Now, it's clear that we will be picking and choosing which message types should be supported. I have implemented your suggestion but have left the old code commented out in case the code ends up being too verbose in the future once we support more message types.

You clearly took the time to read & understand the code and make insightful comments, so thank you!